process/smartContract/process.go

```diff
@@ -1340,6 +1360,32 @@ func (sc *scProcessor) EpochConfirmed(epoch uint32) {
 	log.Debug("scProcessor: penalized too much gas", "enabled", sc.flagPenalizedTooMuchGas.IsSet())
 }
 
+func (sc *scProcessor) handleAsyncStepGas(input *vmcommon.ContractCallInput) (uint64, error) {
+	if input.CallType != vmcommon.AsynchronousCall {
+		return 0, nil
+	}
+
+	asyncCallStepCost := sc.gasSchedule["ElrondAPICost"]["AsyncCallStep"]
```

Comment:
magic strings. maybe keep a simple gasSchedule map[string]uint64 for the scProcessor in special. And save the needed values there.

Reply:
Changed `ArgNewSmartContractProcessor` to accept a `GasSchedule`, and `NewSmartContractProcessor(args)` will only keep the values of interest, discarding the `GasSchedule`.

Magic strings moved into `core/constants.go`.